ImageD11/sinograms/dataset.py

```diff
@@ -155,12 +220,16 @@ def import_scans(self, scans=None, hname = None):
                          (self.detector in hin[scan]['measurement'])) ]
             goodscans = []
             for scan in scans:
-                frames = hin[scan]['measurement'][self.detector]
-                if len(frames.shape)==3: # need 1D series of frames
-                    goodscans.append(scan)
-                    frames_per_scan.append( frames.shape[0] )
-                else:
+                try:
+                    frames = hin[scan]['measurement'][self.detector]
+                    if len(frames.shape)==3: # need 1D series of frames
+                        goodscans.append(scan)
+                        frames_per_scan.append( frames.shape[0] )
+                    else:
+                        print('Bad scan', scan)
+                except KeyError:
```

Comment:
I prefer this as: `if self.detector in scan:` rather than try/except if that is the intention.